File: ncarg2d/man/autograph/agbnch.m
.TH AGBNCH 3NCARG "March 1993" UNIX "NCAR GRAPHICS"
.na
.nh
.SH NAME
AGBNCH - 
Provides an easy way to convert binary dash patterns into
character dash patterns.
.SH SYNOPSIS
AGBNCH (IDSH)
.SH C-BINDING SYNOPSIS
#include <ncarg/ncargC.h>
.sp
char *c_agbnch(int idsh)
.SH DESCRIPTION 
.IP IDSH 12
(an input expression of type INTEGER) is between 0 and
65535, inclusive.
.SH C-BINDING DESCRIPTION
The C-binding argument description is the same as the FORTRAN 
argument description.
.SH ACCESS 
To use AGBNCH or c_agbnch, load the NCAR Graphics libraries ncarg, ncarg_gks, 
and ncarg_c, preferably in that order.  To get smoother curves, 
drawn using spline interpolation, also load libdashsmth.o.  Or,
you can use the ncargf77 command to compile your program and load 
the above libraries, then, to get smoother curves, use the 
-dashsmth option.
.SH SEE ALSO
Online:
autograph,
agback,
agchax,
agchcu,
agchil,
agchnl,
agcurv,
agdshn,
aggetc,
aggetf,
aggeti,
aggetp,
aggetr,
agpwrt,
agrstr,
agsave,
agsetc,
agsetf,
agseti,
agsetp,
agsetr,
agstup,
agutol,
anotat,
displa,
ezmxy,
ezmy,
ezxy,
ezy
.sp
Hardcopy:
NCAR Graphics Fundamentals, UNIX Version
.SH COPYRIGHT
Copyright (C) 1987-2009
.br
University Corporation for Atmospheric Research
.br
The use of this Software is governed by a License Agreement.

File: ncarg2d/src/libncarg/autogrph/agbnch.f
C
C $Id: agbnch.f,v 1.6 2008-07-27 00:14:33 haley Exp $
C
C                Copyright (C)  2000
C        University Corporation for Atmospheric Research
C                All Rights Reserved
C
C The use of this Software is governed by a License Agreement.
C
      CHARACTER*16 FUNCTION AGBNCH (IDSH)
C
C The value of this function is the character-dash-pattern equivalent of
C the integer dash pattern IDSH, a string of quotes and/or dollar signs.
C Note that the support routines IAND and ISHIFT are used.
C
      KDSH=IDSH
C
      DO 101 I=16,1,-1
        IF (IAND(KDSH,1).EQ.0) THEN
          AGBNCH(I:I)=''''
        ELSE
          AGBNCH(I:I)='$'
        END IF
        KDSH=ISHIFT(KDSH,-1)
  101 CONTINUE
C
      RETURN
C
      END
